articles/execute_process_refactor.md

```diff
@@ -0,0 +1,597 @@
+
+### launch_ros.descriptions.Node
+
+This class would represent only the common information required to define a ROS2 node.
+It would not inherit from any other class, and would have a subset of the fields of the current `launch_ros.actions.Node`.
+
+#### Constructor
+
+These parameters are drawn from the shared parameters of the current `launch_ros.actions.Node` and `launch_ros.descriptions.ComposableNode`, and introduces a new collection of `traits`, described in following classes.
+
+| Argument       | Description                                                  |
+| -------------- | ------------------------------------------------------------ |
+| node_name      | Name the node should have                                    |
+| node_namespace | Namespace the node should create topics/services/etc in      |
+| parameters     | List of either paths to yaml files or dictionaries of parameters |
+| remappings     | List of from/to pairs for remapping names                    |
+| arguments      | Container specific arguments to be passed to the node        |
+| traits      | Additional features of the node, specifically Lifecycle or Composable information|
+
+#### Properties
+
+Accessors would be provided for the various constructor parameters: `node_name`, `node_namespace`, `parameters`, `remappings`, `arguments`, and `traits`.
+
+#### Methods
+
+In the current classes, substitutions are handled through internal method calls.
+To support better isolation of class functionality, substitutions will be performed in this class, but exposed from a public method which can be overridden by subclasses to perform additional appropriate substitutions.
+
+| Name    | Description                                                  |
+| ------- | ------------------------------------------------------------ |
+| prepare | Takes a given LaunchContext and Action, and performs actions necessary to prepare the node for execution in that context. This will primarily consist of expanding various substitutions, but may additionally add event handlers related to the Action which will be invoked. Additionally, this will call the `prepare()` method of any defined traits. |
+
+This function would match that of the current `launch_ros.actions.Node` internal method `_perform_substitutions`.
+
+#### Events
+
+No events would be handled or emitted.
+
+### launch_ros.descriptions.ComposableNode
+
+This class would extend the `launch_ros.descriptions.Node` class, and provide the additional information required for defining a node which can be launched in a composable context. 
+
+#### Constructor
+
+Most parameters would be passed to the superclass.
+
+|Argument|Description|
+|---|---|
+| package        | Optional. Name of the ROS package the node plugin lives in. If not specified, the package of the containing executable will be assumed.         |
```

Comment:
@roger-strain I like the feature, but when loading a composable node into an already running executable, you may not necessarily have access to package names. So we should probably mention that this deduction may fail.

Reply:
I'm adding a note to that effect, but as I think about it I worry that we may not have that information more often than not. Might have to determine whether this is feasible at implementation time.